specification/migrate/resource-manager/Microsoft.OffAzure/stable/2020-01-01/migrate.json

```diff
@@ -0,0 +1,4943 @@
+      "post": {
+        "tags": [
+          "Sites"
+        ],
+        "summary": "Method to get site health summary.",
+        "operationId": "Sites_GetSiteHealthSummary",
+        "consumes": [
+          
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "subscriptionId",
+            "in": "path",
+            "description": "The ID of the target subscription.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of the resource group. The name is case insensitive.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "siteName",
+            "in": "path",
+            "description": "Site name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "description": "The API version to use for this operation.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SiteHealthSummaryCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get VMware site health summary.": {
+            "$ref": "./examples/VMwareSiteHealthSummary_Post.json"
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OffAzure/VMwareSites/{siteName}/vcenters/{vcenterName}": {
+      "get": {
+        "tags": [
+          "VCenter"
+        ],
+        "summary": "Method to get a vCenter.",
+        "operationId": "VCenter_GetVCenter",
+        "consumes": [
+          
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "subscriptionId",
+            "in": "path",
+            "description": "The ID of the target subscription.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of the resource group. The name is case insensitive.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "siteName",
+            "in": "path",
+            "description": "Site name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "vcenterName",
+            "i
```

Comment:
This object is fairly large and will likely have many empty collections. I think it would make more sense to use a single heterogeneous array instead of an object exposing multiple homogeneous arrays.
You can do this using discriminators in swagger, basically a single property on the array item would say what type of item it is. The discriminator field would then dictate the schema of the item itself.
If you search this repo for 'discriminator' you'll see tons of examples of RPs doing this, here is one:

The base type (something you'd reference from *MachineProperties): 
https://github.com/Azure/azure-rest-api-specs/blob/08a05190ae96f0f2c83ecbc99bf74322b5afd477/specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/firewallPolicy.json#L670

The discriminated type (types would include 'applications', 'webApplications', 'features'...):
https://github.com/Azure/azure-rest-api-specs/blob/08a05190ae96f0f2c83ecbc99bf74322b5afd477/specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/firewallPolicy.json#L726